test/docker-osx-dev.bats

```diff
@@ -417,6 +417,12 @@ export NEW_ENV_VARIABLE_2=VALUE2"
   assert_output '/some/path'
 }
 
+@test "find_path_to_sync_parent should match paths with weird characters" {
+  export PATHS_TO_SYNC='/some/path2()$HI'
```

Comment:
Thanks for adding a unit test. I can see how a dollar sign would get interpolated. Do parens get interpolated too?